### ISSUE TITLE
zig clang failing to locate stdlib headers

### ISSUE THREAD

**Issue by danielchasehooper** [ziglang/zig#16236]
### Zig Version

0.11.0-dev.3395+1e7dcaa3a

### Steps to Reproduce and Observed Behavior

Happens on macOS, didn't try other OS.

put this in `main.c`
```
#include<assert.h>

int main(int argc, char **argv) {
    assert(argc > 0);
    return 0;
}
```

These succeed
```
$ clang main.c
$ zig cc main.c
```

But this fails
```
$ zig clang main.c
main.c:2:9: fatal error: 'assert.h' file not found
#include<assert.h>
        ^~~~~~~~~~
1 error generated.
```

### Expected Behavior

`zig clang` should work

**Comment by kubkon** [comment]
As in your other issue, might be fixed by #16212

**Comment by kubkon** [comment]
Oh, I noticed only now, you're using `zig clang`. I am not even sure what that should be.

**Comment by kubkon** [comment]
Just checked, this is not fixed by #16212.
